rl4co/models/zoo/ham/policy.py

```diff
@@ -19,31 +23,35 @@ class HeterogeneousAttentionModelPolicy(AutoregressivePolicy):
 
     def __init__(
         self,
-        env_name: str,
         encoder: nn.Module = None,
+        env_name: str = "pdp",
         init_embedding: nn.Module = None,
-        embedding_dim: int = 128,
+        embed_dim: int = 128,
         num_encoder_layers: int = 3,
         num_heads: int = 8,
         normalization: str = "batch",
+        feedforward_hidden: int = 512,
+        sdpa_fn: Optional[Callable] = None,
```

Comment:
[Scaled Dot Product Attention](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html)

We support this since some new versions that are faster such as [FlashAttentionX](https://github.com/Dao-AILab/flash-attention) can be faster